### PR TITLE
docs(PPDSC-2323): add event tracking to toc

### PR DESCRIPTION
**Pull request by mutebg**
PPDSC-2323

**What**

1. Background - to track scroll depths on components pages
2. What did you do - added fireEvent every time a section is in viewport
3. What does the reviewers should expect

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2323-track-scroll-depth
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2323-track-scroll-depth/storybook

File: site/pages/_app.tsx [hunk 1]
```diff
@@ -130,6 +130,8 @@ export default class MyApp extends App<Props, State> {
       instrumentationHandlers.createTealiumTrackHandler(),
     ];
 
+    console.log({pageProps});
```
Comment (Xin00163) [inline site/pages/_app.tsx:133]: still needed?

**Review by baburay23 — APPROVED** [review]

File: site/components/table-of-contents/table-of-contents.tsx [hunk 2]
```diff
@@ -62,15 +87,21 @@ export const TableOfContents: React.FC = () => {
...
-      <Stack flow="vertical-left">
-        {contentsInfo &&
-          contentsInfo.map((info, index) => (
-            <ContentsNavItem key={info.id} itemKey={index} href={`#${info.id}`}>
-              {info.title}
-            </ContentsNavItem>
-          ))}
-      </Stack>
-    </StyledTableOfContents>
+    <InstrumentationProvider context={{area: 'toc-navigation'}}>
```
Comment (Xin00163) [inline site/components/table-of-contents/table-of-contents.tsx:90]: I wondered if I am supposed to see `toc-navigation`
Here is what I got:
```context: {pageOriginator: 'inline-message', pageType: 'ppdsc-2323-track-scroll-depth', area: 'main content', section: 'interactive-demo'}
```

Comment (mutebg) [reply site/components/table-of-contents/table-of-contents.tsx:90]: When you click on the link or when you scroll, these are 2 different events?

when you scroll you should see:
```
{
context: {
   area: 'main content', 
   section: 'anatomy'
},
originator: "page-scroll",
trigger:  "scroll"
}
```

and when you click on the links that is a different event
```
{
context:  {
   area: 'toc-navigation', href: '/ppdsc-2323-track-scroll-depth/components/checkbox/#states', 
   value: 'States'
},
originator:  "link",
trigger: "click",
}
```

File: site/pages/_app.tsx (hunk 1 above)
Comment (Xin00163) [inline site/pages/_app.tsx:133]: Still logging pageProps

File: site/components/table-of-contents/table-of-contents.tsx [hunk 3]
```diff
@@ -27,7 +34,25 @@ export const TableOfContents: React.FC = () => {
...
+
+      if (section) {
+        const data = {
+          originator: 'page-scroll',
+          trigger: EventTrigger.Scroll,
+          context: {
+            section,
+          },
+        };
+        fireEvent(data);
```
Comment (Vanals) [inline site/components/table-of-contents/table-of-contents.tsx:53]: I am a bit unsure why we need this 🤔  I have had a look at the PR website and I can see we already have in place a 'page-scroll' event tracker AND a link event tracker.

TOC is triggered by clicking a link and then scrolls down to the designated area... so we have already event tracking this happening, why do we need an extra event to track this action? an extra set of information? isn't what we are sending enough? feels a bit overkilling. Let me know if I am missing something

Also if must be tracket, I would maybe differentiate this event from the "normal" `page-scrolling` event. I would call it differently to specify this is the event where a TOC section gets selected like 'toc-section-selection', not really the scrolling

**Review by Vanals — CHANGES_REQUESTED** [review]

**Review by Vanals — APPROVED** [review]

**Review by mstuartf — DISMISSED** [review]

File: site/pages/_app.tsx (hunk 1 above)
Comment (mutebg) [reply site/pages/_app.tsx]: thanks, removed

**Review by Xin00163 — APPROVED** [review]

**Review by mstuartf — APPROVED** [review]

**Pull request merged**